File: cs752/RTC/PcClient/PcClient/Core/Background.cs
/*
Author: Daniel Knutsen, Trevor Gerhardt
Date:   Spring 2010
Course: CS 752 
*/

namespace RT {

    public abstract class Background {

        public Color color;

        public void preprocess() { }

        public Color getColor(RenderContext rc, Ray ray) 
        {
            return this.color;
        }

    }

    public class ConstantBackground : Background
    {
        public ConstantBackground(Color color) 
        {
            this.color = color;
        }
    }

} // namespace RT

File: cs752/RTC/Code/Background.cpp
/*
Author: Daniel Knutsen, Trevor Gerhardt
Date:   Spring 2010
Course: CS 752 
*/

#include "Background.h"

#include "Color.h"

namespace RT {

   ConstantBackground::ConstantBackground( Color color ) 
      : color( color ) { }

   void ConstantBackground::preprocess() { }

   Color ConstantBackground::getColor( const RenderContext& rc, const Ray& ray ) const {
      return this->color;
   }

} // namespace RT